library/ssl_misc.h

```diff
@@ -2205,4 +2205,63 @@ static inline int psa_ssl_status_to_mbedtls( psa_status_t status )
 }
 #endif /* MBEDTLS_USE_PSA_CRYPTO || MBEDTLS_SSL_PROTO_TLS1_3 */
 
+/**
+ * \brief       TLS record protection modes
+ */
+typedef enum {
+    MBEDTLS_SSL_MODE_STREAM = 0,
+    MBEDTLS_SSL_MODE_CBC,
+    MBEDTLS_SSL_MODE_CBC_ETM,
+    MBEDTLS_SSL_MODE_AEAD
+} mbedtls_ssl_mode_t;
+
+static inline mbedtls_ssl_mode_t mbedtls_get_mode_from_transform(
```

Comment:
This feels a bit large to be inline, though perhaps it's just because of all the preprocessor conditions and the code isn't that large. Can you measure what the effect of making this inline or not is on the code size? Using a build similar to `all.sh`'s `component_build_arm_none_eabi_gcc_m0plus()`.

Reply:
@mpg You're right, it adds 40bytes of code with PSA, and 48bytes without PSA